src/v/datalake/translation/utils.cc

```diff
@@ -12,8 +12,22 @@
 namespace datalake::translation {
 
 model::offset highest_log_offset_below_next(
-  ss::shared_ptr<storage::log> log, kafka::offset o) {
-    auto next_kafka_offset = kafka::next_offset(o);
+  ss::shared_ptr<storage::log> log, kafka::offset kafka_lto) {
+    /**
+     * If translated offset is smaller than the start offset of the log it means
```

Comment:
Please describe the scenario why this would happen in the comment. During normal operations this shouldn't happen, right?

Reply:
I don't doubt the fix makes it work. I'm afraid it has the opportunity to mask other logic flaws which either already exists or can be introduced in the future.

Reply:
This is strictly related with fast partition movement, in this case we may experience last translated offset pointing into Tiered Storage offset space

Reply:
I thought we'd discussed fixing this from the side of FPM?

Reply:
_Posting these question here so they are visible for anyone else following the conversation and wondering about similar questions. Not blocking anything here._

---

Ah. 🤯 So many questions. So FPM can ignore max collectible offset? Wasn't that an invariant to keep data local? Are we reaching in some direct way to cloud manifest stm to learn what's uploaded?

On the other hand, if max collectible offset is useless now, should we have it defined for this stm (any stm at all really?). Is it only a hint now? Is this functionality overlapping/redundant with space management running?

Reply:
FPM uses archival stm max_collectible offset, i will change it to use the general max collectible offset